doc/user/load_save.md

```diff
@@ -19,13 +22,169 @@ saving data and objects are also available.
+|---------------------------------------------------------------------------------------------------------
+| `Categorical` | `.Categorical()` | bool | Set `true`, if the dataset contains categorical data.        |
+|---------------------------------------------------------------------------------------------------------
+| `Image` | `.Image()` | bool | Set `true`, if we are trying to Load / Save images instead of text formats. |
+|------------------------------------------------------------------------------------------------------------
+| `Model` | `.Model()`| bool | Set `true`, if we are trying to Load / Save an mlpack object                 |
+|         |           |      | (e.g., neural network model)                                                 |
+|-----------------------------------------------------------------------------------------------------------|
+
+
+## Options related to File Formats for (Load / Save)
+
+By default, load/save format is ***autodetected***, but can be manually
+specified with the `format` parameter using one of the options below:
+
+
+|--------------------------------------------------------------------------|
+| Options as a Bitwise | Option as a function name | Type  | Comment       |
+|-----------------------------------------------------------------------------------------------------------|
+| `CSV` | `.FileFormat() = FileType::CSVASCII` | enum | (autodetect extensions `.csv`, `.tsv`): CSV format  |
+|       |                                      |      | with no header.  If loading a sparse matrix and the |
+|       |                                      |      | CSV has three columns, the data is interpreted as a |
+|       |                                      |      | [coordinate list](https://arma.sourceforge.net/docs.html#save_load_mat). |
+|-------------------------------------------------------------------------------------------------------------|
+| `PGM_BIN` | `.FileFormat() = FileType::PGMBinary` | enum | (autodetect extension `.pgm`): PGM image format. |
+|------------------------------------------------------------------------------------------------------------|
+| `PPM_BIN` | `.FileFormat() = FileType::PPMBinary` | enum | (autodetect extension `.ppm`):PPM image format. |
+|------------------------------------------------------------------------------------------------------------|
+| `HDF5_BIN` | `.FileFormat() = FileType::HDF5Binary` | enum | (autodetect extensions `.h5`, `.hdf5`, `.hdf`,
+|            |                                        |       |  `.he5`): [HDF5](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) |
+|            |                                        |       |   binary format; only available if Armadillo is configured with |
+|            |                                        |       | [HDF5 support](https://arma.sourceforge.net/docs.html#config_hpp). |
```

Comment:
Is it necessary to make multiple rows in the table for each entry?  I think that the CSS styling should automatically wrap the columns to keep them from getting too wide... if not, I can make a quick CSS fix.

Reply:
Well to be honest, It is better to have it like this just for the sake of readability in markdown format, nothing else.
I would keep it like this if it is not causing a problem